examples/nextjs/pages/pages/[handle].js

```diff
@@ -1,39 +1,54 @@
-import React from 'react'
-import $nacelle from 'services/nacelle.js'
+import React from 'react';
+import { useRouter } from 'next/router';
+import $nacelle from 'services/nacelle.js';
 
 const Page = ({ page }) => {
-  return page && <pre>{JSON.stringify(page)}</pre>;
+  const router = useRouter();
+
+  // If the page is not yet generated, this will be displayed
+  // initially until getStaticProps() finishes running
+  if (router.isFallback) {
+    return <div>Loading...</div>;
+  }
+
+  return page && <pre>{JSON.stringify(page, null, 2)}</pre>;
 };
 
-export default Page
+export default Page;
 
 export async function getStaticPaths() {
   try {
-    const allContent = await $nacelle.data.allContent()
-    const pages = allContent.filter(entry => entry.type === "page")
+    const allContent = await $nacelle.data.allContent();
```

Comment:
Feels like we're doing the same thing in `getStaticPaths` and `getStaticProps` for each of these. It's probably possible to abstract this into a couple utility functions, no?

Reply:
I think we could, but probably at the expense of clarity. A utility function used in `getStaticPaths` might look like `getPathsFromData({ allData, filterFn })` but to my eyes, that introduces a little bit of a disconnect between the data-fetching in the SDK, and `getStaticPaths` without much benefit.

If I'm not understanding the essence of what you're suggesting, please tell me!

Reply:
I think you're understanding for sure. This seems like a pretty common convention in their examples: https://github.com/vercel/next.js/tree/canary/examples/with-mdx-remote/utils. If we don't want to change it here, that's fine since it's just an example, but I keep thinking about having to come and adjust this at some point when it may not be you or I, and someone having to know that they have to do it in multiple places could potentially cause a bug.

Reply:
Thanks, @johnstonbl01 - that makes sense! I added a utility function to create paths from Nacelle data in https://github.com/getnacelle/nacelle-react/pull/26/commits/7c3f20e531537b7a69941dcae6175e7e6533e66f. I think it keeps it pretty clear after all. There are JSDoc comments to help the merchant dev understand what the utility function is doing.